src/main/java/org/glassfish/enterprise/concurrent/ManagedExecutorServiceImpl.java

```diff
@@ -129,18 +130,22 @@ public ManagedExecutorServiceImpl(String name,
             queue = new ArrayBlockingQueue<>(queueCapacity); 
         }
 
-        threadPoolExecutor = new ManagedThreadPoolExecutor(corePoolSize, maxPoolSize, 
-                keepAliveTime, keepAliveTimeUnit, queue, 
-                this.managedThreadFactory);
-        threadPoolExecutor.setThreadLifeTime(threadLifeTime);
+        if (useForkJoinPool) {

```

Comment:
This is a repetition of the above addition in the other constructor. Perhaps call the other constructor and pass it the queue (additional constructor param) so that you DRY

Reply:
Done!